src/base/tests/test_util.py

```diff
@@ -1364,6 +1369,59 @@ def test_replace_record_references_batch__uniqueness(self):
         [count] = self.env.cr.fetchone()
         self.assertEqual(count, 1)
 
+    @unittest.skipUnless(util.version_gte("18.0"), "Only work on Odoo >= 18")
+    def test_replace_record_references_batch__company_dependent(self):
+        partner_model = self.env["ir.model"].search([("model", "=", "res.partner")])
+        self.env["ir.model.fields"].create(
+            {
+                "name": "x_test_curr",
+                "ttype": "many2one",
+                "model_id": partner_model.id,
+                "relation": "res.currency",
+                "company_dependent": True,
+            }
+        )
+        c1 = self.env["res.currency"].create({"name": "RC1", "symbol": "RC1"})
+        c2 = self.env["res.currency"].create({"name": "RC2", "symbol": "RC2"})
+        c3 = self.env["res.currency"].create({"name": "RC3", "symbol": "RC3"})
+        c4 = self.env["res.currency"].create({"name": "RC4", "symbol": "RC4"})
+
+        p1 = self.env["res.partner"].create({"name": "Captain Jack"})
+        p2 = self.env["res.partner"].create({"name": "River Song"})
+        p3 = self.env["res.partner"].create({"name": "Donna Noble"})
+
+        old = {
+            p1.id: f'{{"1":{c1.id}, "2":{c2.id}, "3":null}}',
+            p2.id: f'{{"1":{c1.id}, "2":{c2.id}, "3":{c3.id}, "4":{c4.id}}}',
+            p3.id: f'{{"1":{c4.id}}}',
+        }
+        for id, value in old.items():
+            self.env.cr.execute(
+                """
+                UPDATE res_partner
+                  SET x_test_curr = %s
+                WHERE id = %s
+                """,
+                [value, id],
+            )
+        mapping = {
+            c1.id: c2.id,
+            c2.id: c3.id,
+            c3.id: c1.id,
+        }
+        add_test_trigger(self.env.cr, "replace_record_references", "res_partner", "UPDATE", f"new.id = {p3.id}")
+        util.replace_record_references_batch(self.env.cr, mapping, "res.currency")
```

Comment:
This could be context manager
```suggestion
        with self.assertNotUpdated("res_partner", [p3.id]):
            util.replace_record_references_batch(self.env.cr, mapping, "res.currency")
```

Signature: `def assertNotUpdated(self, table, ids=None):`.
When ids is None, the whole should be handled, which mean no `INSERT`.

We can also add the positive version `def assertUpdated(self, table, ids=None)`

These functions need to be added in a dedicated commit.

Reply:
To be consistent with the other `assertX` functions, the signature needs to take a `msg` argument.

`assertNotUpdated(self, table, ids=None, msg=None)`